### PR TITLE
use "export" to expose TOPDIR to all child make instead of passing it around every time

### PR DESCRIPTION
**Pull request by v01d**
## Summary

The "export" keyword can be used in make to have all child `$(MAKE)` calls have the variable defined without having to manually pass it via command line. Since `TOPDIR` is needed everywhere, this removes the need to pass it around all the time.

## Impact

Build

## Testing

Built locally

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM, does anyone have more comment?

**Comment by xiaoxiang781216** [comment]
BTW, @v01d should we expose APPDIR/TOPDIR for Makefile  in apps folder?

**Comment by patacongo** [comment]
I suspect this change will add more breakage to the Windows native build.  'export' is a GNU/Bash command and Bash is not used (the shell is CMD.com in the Windows native build).

In the Windows native build, we do use GNUWin32 which provides a small subset of GNU commands, but I can find no reference to the 'export' command being supported.

**Comment by v01d** [comment]
> I suspect this change will add more breakage to the Windows native build. 'export' is a GNU/Bash command and Bash is not used (the shell is CMD.com in the Windows native build).
> 
> In the Windows native build, we do use GNUWin32 which provides a small subset of GNU commands, but I can find no reference to the 'export' command being supported.

But "export" is a Makefile keyword, it should not depend on bash: https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html

**Comment by patacongo** [comment]
> But "export" is a Makefile keyword, it should not depend on bash: https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html

Then I retract my question.

**Comment by v01d** [comment]
> BTW, @v01d should we expose APPDIR/TOPDIR for Makefile in apps folder?

I have this ready, once this is merged I can submit it so that CI runs against this.

**Comment by xiaoxiang781216** [comment]
LGTM, let's merge it to unblock other upcoming PR.

**Pull request merged**
